### PR TITLE
Fix NPEs in DefaultRequestManager

### PR DESCRIPTION
**Pull request by carymrobbins**
Fixes more NPEs due to nullable Boolean values from lsp4j.

Side note - is assuming `false` as the default ok in these cases? Or is this actually a bug with us not initializing the fields correctly?

### CODE REVIEW

File: src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java [hunk 1]
```diff
@@ -397,8 +398,12 @@ public void didClose(DidCloseTextDocumentParams params) {
...
         if (checkStatus()) {
             try {
-                return (serverCapabilities.getCompletionProvider() != null && serverCapabilities.getCompletionProvider()
-                        .getResolveProvider()) ? textDocumentService.resolveCompletionItem(unresolved) : null;
+                return (
+                    Optional.of(serverCapabilities.getCompletionProvider())
+                        .map(x -> x.getResolveProvider())
+                        .orElse(false)
+                ) ? textDocumentService.resolveCompletionItem(unresolved)
+                  : null;
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:406]:
```suggestion
                return (Optional.of(serverCapabilities.getCompletionProvider())
                    .map(x -> x.getResolveProvider()).orElse(false)) ?
                    textDocumentService.resolveCompletionItem(unresolved) : null;
```

File: src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java [hunk 2]
```diff
@@ -411,7 +416,10 @@ public void didClose(DidCloseTextDocumentParams params) {
     public CompletableFuture<Hover> hover(TextDocumentPositionParams params) {
         if (checkStatus()) {
             try {
-                return (serverCapabilities.getHoverProvider()) ? textDocumentService.hover(params) : null;
+                return (
+                    Optional.of(serverCapabilities.getHoverProvider()).orElse(false)
+                ) ? textDocumentService.hover(params)
+                  : null;
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:422]:
```suggestion
                return (
                    Optional.of(serverCapabilities.getHoverProvider()).orElse(false)) ?
                        textDocumentService.hover(params) : null;
               
```

File: src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java [hunk 3]
```diff
@@ -397,8 +398,12 @@ public void didClose(DidCloseTextDocumentParams params) {
     public CompletableFuture<CompletionItem> resolveCompletionItem(CompletionItem unresolved) {
         if (checkStatus()) {
             try {
-                return (serverCapabilities.getCompletionProvider() != null && serverCapabilities.getCompletionProvider()
-                        .getResolveProvider()) ? textDocumentService.resolveCompletionItem(unresolved) : null;
+                return (
+                    Optional.of(serverCapabilities.getCompletionProvider())
+                        .map(x -> x.getResolveProvider())
```
Comment (nixel2007) [inline src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:403]: `.map(CompletionProvider::getResolveProvider)`

and so on... @NipunaRanasinghe WDYT?

Comment (carymrobbins) [reply src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:403]: I intentionally used a lambda instead as it reads a bit easier to me. Doesn't seem valuable to make things mire verbose than necessary IMHO.

Comment (nixel2007) [reply src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:403]: `::` syntax also provides info about type of `x`, but I understand your point.

Comment (NipunaRanasinghe) [reply src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java:403]: > `.map(CompletionProvider::getResolveProvider)`
> 
> and so on... @NipunaRanasinghe WDYT?

@nixel2007 @matthewmichihara I'm being 0 here as the 2 approaches don't have a performance different so it's just a matter of readability + personal preference. :)

**Comment by NipunaRanasinghe** [comment]
@carymrobbins Shall we add the requested changes so that we can go ahead and merge this?

**Comment by carymrobbins** [comment]
@NipunaRanasinghe Yes, sorry for the delay. I've went ahead and auto-applied your suggestions.

**Comment by NipunaRanasinghe** [comment]
> @NipunaRanasinghe Yes, sorry for the delay. I've went ahead and auto-applied your suggestions.

Cool!

**Pull request merged**